### PR TITLE
Skeleton class for GpioMonitor

### PR DESCRIPTION
**Pull request by RekhaAparna01**
GPIO monitor class is responsible for gathering information about GPIO pins from the config JSON and instantiates the GpioEventhandler class

### CODE REVIEW

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: include/gpioMonitor.hpp [hunk 1]
```diff
@@ -89,4 +92,44 @@ class GpioEventHandler
...
+  public:
+    GpioMonitor() = delete;
+    ~GpioMonitor() = default;
+    GpioMonitor(const GpioMonitor&) = delete;
+    GpioMonitor& operator=(const GpioMonitor&) = delete;
+    GpioMonitor(GpioMonitor&&) = delete;
+    GpioMonitor& operator=(GpioMonitor&&) = delete;
+
+    /**
+     *@brief constructor
```
[the remark this answers is not in the document]
Comment (RekhaAparna01) [reply include/gpioMonitor.hpp:106]: Updated

File: src/manager.cpp [hunk 2]
```diff
@@ -184,6 +185,9 @@ void Manager::SetTimerToDetectVpdCollectionStatus()
 
             const nlohmann::json& l_sysCfgJsonObj =
                 m_worker->getSysCfgJsonObj();
+
+            GpioMonitor l_GpioMonitorObj(l_sysCfgJsonObj, m_ioContext);
```
[the remark this answers is not in the document]
Comment (RekhaAparna01) [reply src/manager.cpp:189]: This is for instantiating the GpioMonitor Class. I was mistaken and initiated in the wrong way.

File: src/manager.cpp [hunk 3]
```diff
@@ -39,6 +39,11 @@ Manager::Manager(
 
         // set async timer to detect if VPD collection is done.
         SetTimerToDetectVpdCollectionStatus();
+
+        const nlohmann::json& l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
```
Comment (SunnySrivastava1984) [inline src/manager.cpp:43]: Don't need this variable as it is not being used at multiple places. Please get this in place.

File: include/gpioMonitor.hpp [hunk 4]
```diff
@@ -89,4 +92,44 @@ class GpioEventHandler
...
+  private:
+    /**
+     * @brief API to extract GPIO information from JSON.
+     *
+     * This API will extract the gpio information from system config JSON
+     * and store it in GpioEventHandler's private variables.
+     *
+     * @param[in] i_ioContext - Pointer to io context object.
+     */
+    void initGpioInfos(
```
Comment (SunnySrivastava1984) [inline include/gpioMonitor.hpp:127]: initHandlerForGpio.
As event handler will be initialized in the function.

File: include/gpioMonitor.hpp [hunk 5]
```diff
@@ -89,4 +92,44 @@ class GpioEventHandler
...
+    GpioMonitor(const nlohmann::json& i_sysCfgJsonObj,
+                const std::shared_ptr<boost::asio::io_context>& i_ioContext) :
+        m_sysCfgJsonObj(i_sysCfgJsonObj)
+    {
+        initGpioInfos(i_ioContext);
+    }
+
+  private:
+    /**
+     * @brief API to extract GPIO information from JSON.
```
Comment (SunnySrivastava1984) [inline include/gpioMonitor.hpp:120]: Brief is not aligned with what is intended to be done in the function.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: include/gpioMonitor.hpp (hunk 5 above)
Comment (RekhaAparna01) [reply include/gpioMonitor.hpp]: Updated

File: include/gpioMonitor.hpp (hunk 4 above)
Comment (RekhaAparna01) [reply include/gpioMonitor.hpp]: Updated

File: src/manager.cpp (hunk 3 above)
Comment (RekhaAparna01) [reply src/manager.cpp]: Updated

File: include/gpioMonitor.hpp [hunk 6]
```diff
@@ -89,4 +92,44 @@ class GpioEventHandler
...
+     * @brief constructor
+     *
+     * @param[in] i_sysCfgJsonObj - System config JSON Object.
+     * @param[in] i_ioContext - pointer to io context object.
+     */
+    GpioMonitor(const nlohmann::json i_sysCfgJsonObj,
+                const std::shared_ptr<boost::asio::io_context>& i_ioContext) :
+        m_sysCfgJsonObj(i_sysCfgJsonObj)
+    {
+        initHandlerForGpio(i_ioContext);
```
Comment (SunnySrivastava1984) [inline include/gpioMonitor.hpp:115]: This api needs sysConfigJSon to work. Its better if we have a check for json empty before calling this API. Else just return

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: include/gpioMonitor.hpp (hunk 6 above)
Comment (RekhaAparna01) [reply include/gpioMonitor.hpp]: updated

**Review by SunnySrivastava1984 — APPROVED** [review]

File: include/manager.hpp [hunk 7]
```diff
@@ -240,6 +241,9 @@ class Manager
 
     // Shared pointer to worker class
     std::shared_ptr<Worker> m_worker;
+
+    // shared pointer to GpioMonitor class
```
Comment (branupama) [inline include/manager.hpp:245]: Shared

**Review by branupama — CHANGES_REQUESTED** [review]

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**